server/src/main/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponse.java

```diff
@@ -71,5 +118,121 @@ public void writeTo(StreamOutput out) throws IOException {
+        parser.nextToken(); //go to settings object itself
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+
+        switch (settingsFieldName) {
+            case "settings":
+                indexToSettings.put(currentIndexName, Settings.fromXContent(parser));
+                break;
+            case "defaults":
+                indexToDefaultSettings.put(currentIndexName, Settings.fromXContent(parser));
+                break;
+            default:
+                //We don't know this field, ignore it
+        }
+    }
+
+    public static GetSettingsResponse fromXContent(XContentParser parser) throws IOException {
+        HashMap<String, Settings> indexToSettings = new HashMap<>();
+        HashMap<String, Settings> indexToDefaultSettings = new HashMap<>();
+
+        if (parser.currentToken() == null) {
+            parser.nextToken();
+        }
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+        parser.nextToken();
+
+        while (!parser.isClosed() && parser.currentToken() == XContentParser.Token.FIELD_NAME) {
+            String indexName = parser.currentName();
+            parser.nextToken(); //go to per-index settings object
+            XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+            parser.nextToken();
+
+            parseSettingsField(parser, indexName, indexToSettings, indexToDefaultSettings);  //this will get one of settings or defaults
+            parser.nextToken();
+
+            if (parser.currentToken() == XContentParser.Token.FIELD_NAME) {
+                //we have both settings and default settings, let's grab the second one
+                parseSettingsField(parser, indexName, indexToSettings, indexToDefaultSettings);
+                parser.nextToken();
+            }
+
+            XContentParserUtils.ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.currentToken(), parser::getTokenLocation);
+            parser.nextToken(); //we should now be positioned at the beginning of the next index's settings, or at the end
+        }
+
+        ImmutableOpenMap<String, Settings> settingsMap = ImmutableOpenMap.<String, Settings>builder().putAll(indexToSettings).build();
+        ImmutableOpenMap<String, Settings> defaultSettingsMap =
+            ImmutableOpenMap.<String, Settings>builder().putAll(indexToDefaultSettings).build();
+
+
+        return new GetSettingsResponse(settingsMap, defaultSettingsMap);
+    }
+
+    @Override
+    public String toString() {
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            XContentBuilder builder = new XContentBuilder(JsonXContent.jsonXContent, baos);
+            toXContent(builder, ToXContent.EMPTY_PARAMS, false);
+            return Strings.toString(builder);
```

Comment:
you could provide the params as `ToXContent.Params`, then you could call the other `Strings.toString` method that accept a `ToXContent` and handles exceptions as well, plus get rid of two different `toXContent` methods.

Reply:
Not sure I follow this comment - none of the Strings.toString() methods accept any Params.